lib/pry-doc.rb

```diff
@@ -56,15 +56,13 @@ def self.receiver_notation_for(meth)
     # @return [Array] The aliases of a method if it exists
     #                 otherwise, return empty array
     def self.aliases(meth)
-      host        = is_singleton?(meth) ? meth.receiver : meth.owner
-      method_type = is_singleton?(meth) ? :method : :instance_method
```

Comment:
`owner` of class method is singleton class. So we can call `#method ` on either case (instance methods or class methods)

```ruby
class Test
  def a
  end

  def b
  end

  class << self
    def c
    end

    def d
    end
  end
end

a_method = Test.instance_method(:a)
c_method = Test.method(:c)

a_method = Test.instance_method(:a)
# => #<UnboundMethod: Test#a>
c_method = Test.method(:c)
# => #<Method: Test.c>
a_method.owner
# => Test
c_method.owner
# => #<Class:Test>
a_method.owner.instance_method(:a)
# => #<UnboundMethod: Test#a>
c_method.owner.instance_method(:c)
# => #<UnboundMethod: #<Class:Test>#c>
```